cmake/version.cmake

```diff
@@ -23,5 +23,11 @@ else()
   set(REVISION_LAST "${REVISION}")
 endif()
 
+if (REVISION MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+)")
+    set(REVISION_MAJOR ${CMAKE_MATCH_1})
+    set(REVISION_MINOR ${CMAKE_MATCH_2})
+    set(REVISION_PATCH ${CMAKE_MATCH_3})
+    message(STATUS "Revision major ${REVISION_MAJOR}, minor ${REVISION_MINOR}, patch ${REVISION_PATCH}")
```

Comment:
With this patch, cmake output looks like
```
-- Revision major 0, minor 24, patch 0
-- Revision is 0.24.0-332cab0e
```
We should combine these two messages to output something like
```
Revision is 0.24.0-332cab0e (major 0, minor 24, patch 0)
```

Reply:
ok, changed. This will now print:
```
-- Revision is 0.24.0-5a196623 (major 0, minor 24, patch 0)
```